# Blockbuster data cleaning 🍿
In this notebook, we'll explore,clean and export the CSV files in `'../data/old_HDD.csv'` to the folder `'../clean_data'`.

To later turn this information into a `MySQL` database.

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
pd.set_option('display.max_columns', None) # Display all columns

## Load the `CSV` into dataframes

In [3]:
df_actor_raw = pd.read_csv('../data/actor.csv')
df_category_raw = pd.read_csv('../data/category.csv')
df_film_raw = pd.read_csv('../data/film.csv')
df_inventory_raw = pd.read_csv('../data/inventory.csv')
df_language_raw = pd.read_csv('../data/language.csv')
df_oldHDD_raw = pd.read_csv('../data/old_HDD.csv')
df_rental_raw = pd.read_csv('../data/rental.csv')

## 1. Cleaning `actors.csv`

#### 1.1 Data Cleaning & Exploration

In [4]:
df_actor = df_actor_raw.copy()
actor_shape = df_actor.shape

In [5]:
df_actor.head()

actor_id first_name     last_name          last_update
0         1   PENELOPE       GUINESS  2006-02-15 04:34:33
1         2       NICK      WAHLBERG  2006-02-15 04:34:33
2         3         ED         CHASE  2006-02-15 04:34:33
3         4   JENNIFER         DAVIS  2006-02-15 04:34:33
4         5     JOHNNY  LOLLOBRIGIDA  2006-02-15 04:34:33

In [6]:
df_actor.tail()

actor_id first_name last_name          last_update
195       196       BELA    WALKEN  2006-02-15 04:34:33
196       197      REESE      WEST  2006-02-15 04:34:33
197       198       MARY    KEITEL  2006-02-15 04:34:33
198       199      JULIA   FAWCETT  2006-02-15 04:34:33
199       200      THORA    TEMPLE  2006-02-15 04:34:33

In [7]:
df_actor.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   actor_id     200 non-null    int64 
 1   first_name   200 non-null    object
 2   last_name    200 non-null    object
 3   last_update  200 non-null    object
dtypes: int64(1), object(3)
memory usage: 6.4+ KB


In [8]:
df_actor.describe()

actor_id
count  200.000000
mean   100.500000
std     57.879185
min      1.000000
25%     50.750000
50%    100.500000
75%    150.250000
max    200.000000

In [9]:
df_actor.isna().any()

actor_id       False
first_name     False
last_name      False
last_update    False
dtype: bool

In [10]:
df_actor.drop_duplicates()
df_actor.shape

(200, 4)

In [11]:
df_actor.last_update.value_counts()

last_update
2006-02-15 04:34:33    200
Name: count, dtype: int64

In [12]:
df_actor.last_update.unique() # They are all the same datetime value, not useful info

array(['2006-02-15 04:34:33'], dtype=object)

In [13]:
df_actor.drop(columns='last_update', inplace=True)

In [14]:
df_actor

actor_id first_name     last_name
0           1   PENELOPE       GUINESS
1           2       NICK      WAHLBERG
2           3         ED         CHASE
3           4   JENNIFER         DAVIS
4           5     JOHNNY  LOLLOBRIGIDA
..        ...        ...           ...
195       196       BELA        WALKEN
196       197      REESE          WEST
197       198       MARY        KEITEL
198       199      JULIA       FAWCETT
199       200      THORA        TEMPLE

[200 rows x 3 columns]

In [15]:
df_actor.to_csv('../clean_data/actor_clean.csv', index=0)

### The `actor.csv` file is good.

## 2. Cleaning `category.csv`

#### 2.1 Data Exploration & Cleaning

In [16]:
df_category = df_category_raw.copy()
category_shape = df_category.shape
category_shape

(16, 3)

In [17]:
df_category

category_id         name          last_update
0             1       Action  2006-02-15 04:46:27
1             2    Animation  2006-02-15 04:46:27
2             3     Children  2006-02-15 04:46:27
3             4     Classics  2006-02-15 04:46:27
4             5       Comedy  2006-02-15 04:46:27
5             6  Documentary  2006-02-15 04:46:27
6             7        Drama  2006-02-15 04:46:27
7             8       Family  2006-02-15 04:46:27
8             9      Foreign  2006-02-15 04:46:27
9            10        Games  2006-02-15 04:46:27
10           11       Horror  2006-02-15 04:46:27
11           12        Music  2006-02-15 04:46:27
12           13          New  2006-02-15 04:46:27
13           14       Sci-Fi  2006-02-15 04:46:27
14           15       Sports  2006-02-15 04:46:27
15           16       Travel  2006-02-15 04:46:27

In [18]:
df_category.drop(columns='last_update', inplace=True)

In [19]:
df_category.head()

category_id       name
0            1     Action
1            2  Animation
2            3   Children
3            4   Classics
4            5     Comedy

The `category.csv` is good, let's export.

In [20]:
df_category.to_csv('../clean_data/category_clean.csv', index=0)

## 3. Cleaning `film.csv`

#### 3.1 Exploring the data

In [21]:
df_film = df_film_raw.copy()
film_shape = df_film.shape
film_shape

(1000, 13)

In [22]:
df_film.head()

film_id             title  \
0        1  ACADEMY DINOSAUR   
1        2    ACE GOLDFINGER   
2        3  ADAPTATION HOLES   
3        4  AFFAIR PREJUDICE   
4        5       AFRICAN EGG   

                                         description  release_year  \
0  A Epic Drama of a Feminist And a Mad Scientist...          2006   
1  A Astounding Epistle of a Database Administrat...          2006   
2  A Astounding Reflection of a Lumberjack And a ...          2006   
3  A Fanciful Documentary of a Frisbee And a Lumb...          2006   
4  A Fast-Paced Documentary of a Pastry Chef And ...          2006   

   language_id  original_language_id  rental_duration  rental_rate  length  \
0            1                   NaN                6         0.99      86   
1            1                   NaN                3         4.99      48   
2            1                   NaN                7         2.99      50   
3            1                   NaN                5         2.99     117   
4            1                   NaN                6         2.99     130   

   replacement_cost rating                  special_features  \
0             20.99     PG  Deleted Scenes,Behind the Scenes   
1             12.99      G           Trailers,Deleted Scenes   
2             18.99  NC-17           Trailers,Deleted Scenes   
3             26.99      G    Commentaries,Behind the Scenes   
4             22.99      G                    Deleted Scenes   

           last_update  
0  2006-02-15 05:03:42  
1  2006-02-15 05:03:42  
2  2006-02-15 05:03:42  
3  2006-02-15 05:03:42  
4  2006-02-15 05:03:42

In [23]:
df_film.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 13 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   film_id               1000 non-null   int64  
 1   title                 1000 non-null   object 
 2   description           1000 non-null   object 
 3   release_year          1000 non-null   int64  
 4   language_id           1000 non-null   int64  
 5   original_language_id  0 non-null      float64
 6   rental_duration       1000 non-null   int64  
 7   rental_rate           1000 non-null   float64
 8   length                1000 non-null   int64  
 9   replacement_cost      1000 non-null   float64
 10  rating                1000 non-null   object 
 11  special_features      1000 non-null   object 
 12  last_update           1000 non-null   object 
dtypes: float64(3), int64(5), object(5)
memory usage: 101.7+ KB


In [24]:
df_film.describe()

film_id  release_year  language_id  original_language_id  \
count  1000.000000        1000.0       1000.0                   0.0   
mean    500.500000        2006.0          1.0                   NaN   
std     288.819436           0.0          0.0                   NaN   
min       1.000000        2006.0          1.0                   NaN   
25%     250.750000        2006.0          1.0                   NaN   
50%     500.500000        2006.0          1.0                   NaN   
75%     750.250000        2006.0          1.0                   NaN   
max    1000.000000        2006.0          1.0                   NaN   

       rental_duration  rental_rate       length  replacement_cost  
count      1000.000000  1000.000000  1000.000000       1000.000000  
mean          4.985000     2.980000   115.272000         19.984000  
std           1.411654     1.646393    40.426332          6.050833  
min           3.000000     0.990000    46.000000          9.990000  
25%           4.000000     0.990000    80.000000         14.990000  
50%           5.000000     2.990000   114.000000         19.990000  
75%           6.000000     4.990000   149.250000         24.990000  
max           7.000000     4.990000   185.000000         29.990000

#### 3.2 Data Cleaning

In [25]:
df_film.original_language_id.isnull().sum()

1000

The colummn is 100% `NaN`, with this information, let's drop it.

In [26]:
df_film.drop(columns=['last_update', 'original_language_id'] , inplace=True)

In [27]:
df_film.special_features.value_counts()

special_features
Trailers,Commentaries,Behind the Scenes                   79
Trailers                                                  72
Trailers,Commentaries                                     72
Trailers,Behind the Scenes                                72
Deleted Scenes,Behind the Scenes                          71
Commentaries,Behind the Scenes                            70
Behind the Scenes                                         70
Trailers,Deleted Scenes                                   66
Commentaries,Deleted Scenes,Behind the Scenes             66
Commentaries,Deleted Scenes                               65
Trailers,Commentaries,Deleted Scenes                      64
Commentaries                                              62
Deleted Scenes                                            61
Trailers,Commentaries,Deleted Scenes,Behind the Scenes    61
Trailers,Deleted Scenes,Behind the Scenes                 49
Name: count, dtype: int64

In [28]:

features = ['Deleted Scenes', 'Behind the Scenes', 'Trailers', 'Commentaries']

for feature in features:
    df_film[feature] = df_film['special_features'].str.contains(feature).map({True: 'Y', False: 'N'})

# Drop original column
df_film.drop(columns='special_features', inplace=True)
df_film.head()


film_id             title  \
0        1  ACADEMY DINOSAUR   
1        2    ACE GOLDFINGER   
2        3  ADAPTATION HOLES   
3        4  AFFAIR PREJUDICE   
4        5       AFRICAN EGG   

                                         description  release_year  \
0  A Epic Drama of a Feminist And a Mad Scientist...          2006   
1  A Astounding Epistle of a Database Administrat...          2006   
2  A Astounding Reflection of a Lumberjack And a ...          2006   
3  A Fanciful Documentary of a Frisbee And a Lumb...          2006   
4  A Fast-Paced Documentary of a Pastry Chef And ...          2006   

   language_id  rental_duration  rental_rate  length  replacement_cost rating  \
0            1                6         0.99      86             20.99     PG   
1            1                3         4.99      48             12.99      G   
2            1                7         2.99      50             18.99  NC-17   
3            1                5         2.99     117             26.99      G   
4            1                6         2.99     130             22.99      G   

  Deleted Scenes Behind the Scenes Trailers Commentaries  
0              Y                 Y        N            N  
1              Y                 N        Y            N  
2              Y                 N        Y            N  
3              N                 Y        N            Y  
4              Y                 N        N            N

The rest of the dataframe is clean

In [29]:
df_category.to_csv('../clean_data/film_clean.csv', index=0)

## 4. Cleaning `inventory.csv`

#### 4.1 Exploring the data

In [30]:
df_inventory = df_inventory_raw.copy()
inventory_shape = df_inventory.shape
inventory_shape

(1000, 4)

In [31]:
df_inventory.head()

inventory_id  film_id  store_id          last_update
0             1        1         1  2006-02-15 05:09:17
1             2        1         1  2006-02-15 05:09:17
2             3        1         1  2006-02-15 05:09:17
3             4        1         1  2006-02-15 05:09:17
4             5        1         2  2006-02-15 05:09:17

In [32]:
df_inventory.tail()

inventory_id  film_id  store_id          last_update
995           996      222         2  2006-02-15 05:09:17
996           997      222         2  2006-02-15 05:09:17
997           998      222         2  2006-02-15 05:09:17
998           999      223         2  2006-02-15 05:09:17
999          1000      223         2  2006-02-15 05:09:17

In [33]:
df_inventory.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   inventory_id  1000 non-null   int64 
 1   film_id       1000 non-null   int64 
 2   store_id      1000 non-null   int64 
 3   last_update   1000 non-null   object
dtypes: int64(3), object(1)
memory usage: 31.4+ KB


In [34]:
df_inventory.describe()

inventory_id      film_id     store_id
count   1000.000000  1000.000000  1000.000000
mean     500.500000   109.866000     1.497000
std      288.819436    63.862042     0.500241
min        1.000000     1.000000     1.000000
25%      250.750000    56.000000     1.000000
50%      500.500000   111.500000     1.000000
75%      750.250000   164.000000     2.000000
max     1000.000000   223.000000     2.000000

#### 4.2 Data Cleaning

inventory_shape

In [35]:
inventory_shape

(1000, 4)

In [36]:
df_inventory.isnull().any()

inventory_id    False
film_id         False
store_id        False
last_update     False
dtype: bool

In [37]:
df_inventory.drop_duplicates()

inventory_id  film_id  store_id          last_update
0               1        1         1  2006-02-15 05:09:17
1               2        1         1  2006-02-15 05:09:17
2               3        1         1  2006-02-15 05:09:17
3               4        1         1  2006-02-15 05:09:17
4               5        1         2  2006-02-15 05:09:17
..            ...      ...       ...                  ...
995           996      222         2  2006-02-15 05:09:17
996           997      222         2  2006-02-15 05:09:17
997           998      222         2  2006-02-15 05:09:17
998           999      223         2  2006-02-15 05:09:17
999          1000      223         2  2006-02-15 05:09:17

[1000 rows x 4 columns]

In [38]:
df_inventory.drop(columns='last_update', inplace=True)

In [39]:
df_inventory

inventory_id  film_id  store_id
0               1        1         1
1               2        1         1
2               3        1         1
3               4        1         1
4               5        1         2
..            ...      ...       ...
995           996      222         2
996           997      222         2
997           998      222         2
998           999      223         2
999          1000      223         2

[1000 rows x 3 columns]

In [40]:
df_category.to_csv('../clean_data/inventory_clean.csv', index=0)

## 5. Cleaning `language.csv`

#### 5.1 Data Exploration

In [41]:
df_language = df_language_raw.copy()
language_shape = df_language.shape
language_shape

(6, 3)

In [42]:
df_language

language_id      name          last_update
0            1   English  2006-02-15 05:02:19
1            2   Italian  2006-02-15 05:02:19
2            3  Japanese  2006-02-15 05:02:19
3            4  Mandarin  2006-02-15 05:02:19
4            5    French  2006-02-15 05:02:19
5            6    German  2006-02-15 05:02:19

#### 5.2 Data Cleaning

Let's proceed to clean language table, remove `last_update` and move on.

In [43]:
df_language.drop(columns='last_update', inplace=True)

In [44]:
df_language

language_id      name
0            1   English
1            2   Italian
2            3  Japanese
3            4  Mandarin
4            5    French
5            6    German

In [45]:
df_language.to_csv('../clean_data/language_clean.csv', index= 0)

## 6. Cleaning `old_HDD.csv`

#### 6.1 Data Exploration

In [46]:
df_oldHDD = df_oldHDD_raw.copy()
oldHDD_shape = df_oldHDD.shape
oldHDD_shape

(1000, 5)

In [47]:
df_oldHDD.head()

first_name last_name                  title  release_year  category_id
0   PENELOPE   GUINESS       ACADEMY DINOSAUR          2006            6
1   PENELOPE   GUINESS   ANACONDA CONFESSIONS          2006            2
2   PENELOPE   GUINESS            ANGELS LIFE          2006           13
3   PENELOPE   GUINESS  BULWORTH COMMANDMENTS          2006           10
4   PENELOPE   GUINESS          CHEAPER CLYDE          2006           14

In [48]:
df_film.head()

film_id             title  \
0        1  ACADEMY DINOSAUR   
1        2    ACE GOLDFINGER   
2        3  ADAPTATION HOLES   
3        4  AFFAIR PREJUDICE   
4        5       AFRICAN EGG   

                                         description  release_year  \
0  A Epic Drama of a Feminist And a Mad Scientist...          2006   
1  A Astounding Epistle of a Database Administrat...          2006   
2  A Astounding Reflection of a Lumberjack And a ...          2006   
3  A Fanciful Documentary of a Frisbee And a Lumb...          2006   
4  A Fast-Paced Documentary of a Pastry Chef And ...          2006   

   language_id  rental_duration  rental_rate  length  replacement_cost rating  \
0            1                6         0.99      86             20.99     PG   
1            1                3         4.99      48             12.99      G   
2            1                7         2.99      50             18.99  NC-17   
3            1                5         2.99     117             26.99      G   
4            1                6         2.99     130             22.99      G   

  Deleted Scenes Behind the Scenes Trailers Commentaries  
0              Y                 Y        N            N  
1              Y                 N        Y            N  
2              Y                 N        Y            N  
3              N                 Y        N            Y  
4              Y                 N        N            N

#### 6.2 Data Cleaning

We can add information and merge by title, adding more information to our `film.csv` dataframe, like  `category_id`

In [49]:
# Merge the two DataFrames on the 'title' column to add the 'category_id' column
df_film = df_film.merge(df_oldHDD, on='title', how='left')

In [50]:
df_film.head()

film_id             title  \
0        1  ACADEMY DINOSAUR   
1        1  ACADEMY DINOSAUR   
2        1  ACADEMY DINOSAUR   
3        1  ACADEMY DINOSAUR   
4        2    ACE GOLDFINGER   

                                         description  release_year_x  \
0  A Epic Drama of a Feminist And a Mad Scientist...            2006   
1  A Epic Drama of a Feminist And a Mad Scientist...            2006   
2  A Epic Drama of a Feminist And a Mad Scientist...            2006   
3  A Epic Drama of a Feminist And a Mad Scientist...            2006   
4  A Astounding Epistle of a Database Administrat...            2006   

   language_id  rental_duration  rental_rate  length  replacement_cost rating  \
0            1                6         0.99      86             20.99     PG   
1            1                6         0.99      86             20.99     PG   
2            1                6         0.99      86             20.99     PG   
3            1                6         0.99      86             20.99     PG   
4            1                3         4.99      48             12.99      G   

  Deleted Scenes Behind the Scenes Trailers Commentaries first_name last_name  \
0              Y                 Y        N            N   PENELOPE   GUINESS   
1              Y                 Y        N            N  CHRISTIAN     GABLE   
2              Y                 Y        N            N    LUCILLE     TRACY   
3              Y                 Y        N            N     SANDRA      PECK   
4              Y                 N        Y            N        BOB   FAWCETT   

   release_year_y  category_id  
0          2006.0          6.0  
1          2006.0          6.0  
2          2006.0          6.0  
3          2006.0          6.0  
4          2006.0         11.0

In [51]:
df_film.isna().sum()

film_id                0
title                  0
description            0
release_year_x         0
language_id            0
rental_duration        0
rental_rate            0
length                 0
replacement_cost       0
rating                 0
Deleted Scenes         0
Behind the Scenes      0
Trailers               0
Commentaries           0
first_name           386
last_name            386
release_year_y       386
category_id          386
dtype: int64

This means several things:
- There are `386` titles in the `films` table that `old_HDD` does not contain as we can see by the amount of nulls.
- As there are multiple rows for the same title in the `old_HDD` table (there are multiple actors in one ttile), we have the same amount of duplicates as nulls.
- We will confirm this by searching the title of a row with `null` on category_id

In [52]:
df_film[df_film['title'] == 'AFRICAN EGG']

film_id        title                                        description  \
9        5  AFRICAN EGG  A Fast-Paced Documentary of a Pastry Chef And ...   

   release_year_x  language_id  rental_duration  rental_rate  length  \
9            2006            1                6         2.99     130   

   replacement_cost rating Deleted Scenes Behind the Scenes Trailers  \
9             22.99      G              Y                 N        N   

  Commentaries first_name last_name  release_year_y  category_id  
9            N        NaN       NaN             NaN          NaN

In [53]:
df_oldHDD[df_oldHDD['title'] == 'AFRICAN EGG']

Empty DataFrame
Columns: [first_name, last_name, title, release_year, category_id]
Index: []

As we can see, it does not exist in the `oldHDD` table

In [54]:
df_film.drop(columns=['release_year_y', 'first_name', 'last_name'], inplace=True) # Remove dulicate years with missing info, and actor info
df_film.rename(columns={'release_year_x': 'release_year', 'Deleted Scenes': 'deleted_scenes', 
                        'Behind the Scenes': 'behind_scenes', 'Trailers': 'trailers', 'Commentaries':'commentaries'}, inplace=True)
df_film.fillna('unknown', inplace=True)

/var/folders/f2/ty6sgcw911g2_3vqhhq561vm0000gn/T/ipykernel_23136/3326145887.py:4: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value 'unknown' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.
  df_film.fillna('unknown', inplace=True)


In [55]:
df_film.drop_duplicates(inplace=True)

In [56]:
df_film.shape

(1000, 15)

In [57]:
df_film[df_film['title']=='AFRICAN EGG']

film_id        title                                        description  \
9        5  AFRICAN EGG  A Fast-Paced Documentary of a Pastry Chef And ...   

   release_year  language_id  rental_duration  rental_rate  length  \
9          2006            1                6         2.99     130   

   replacement_cost rating deleted_scenes behind_scenes trailers commentaries  \
9             22.99      G              Y             N        N            N   

  category_id  
9     unknown

- We have added the `category_id` to our films table, adding valuable information for `614` titles, all the ones present in `old_HDD`
- Let's now export the even cleaner `films` table and check if all the actors in `old_HDD` are in `actors`

In [58]:
df_film.to_csv('../clean_data/film_clean.csv', index= 0)

In [59]:
result = df_oldHDD.apply(lambda row: (row['first_name'], row['last_name']) in
                          [(row2['first_name'], row2['last_name']) for index2, row2 in df_actor.iterrows()], axis=1)

if result.all():
    print("All combinations from df_old are present in df_actor.")
else:
    print("Not all combinations from df_old are present in df_actor.")

All combinations from df_old are present in df_actor.


## 7. Cleaning `rental.csv`

#### 7.1 Data Exploration & Cleaning

In [60]:
df_rental = df_rental_raw.copy()
rental_size = df_rental.shape
rental_size

(1000, 7)

In [61]:
df_rental.head()

rental_id          rental_date  inventory_id  customer_id  \
0          1  2005-05-24 22:53:30           367          130   
1          2  2005-05-24 22:54:33          1525          459   
2          3  2005-05-24 23:03:39          1711          408   
3          4  2005-05-24 23:04:41          2452          333   
4          5  2005-05-24 23:05:21          2079          222   

           return_date  staff_id          last_update  
0  2005-05-26 22:04:30         1  2006-02-15 21:30:53  
1  2005-05-28 19:40:33         1  2006-02-15 21:30:53  
2  2005-06-01 22:12:39         1  2006-02-15 21:30:53  
3  2005-06-03 01:43:41         2  2006-02-15 21:30:53  
4  2005-06-02 04:33:21         1  2006-02-15 21:30:53

In [62]:
df_rental.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   rental_id     1000 non-null   int64 
 1   rental_date   1000 non-null   object
 2   inventory_id  1000 non-null   int64 
 3   customer_id   1000 non-null   int64 
 4   return_date   1000 non-null   object
 5   staff_id      1000 non-null   int64 
 6   last_update   1000 non-null   object
dtypes: int64(4), object(3)
memory usage: 54.8+ KB


In [63]:
df_rental.describe()

rental_id  inventory_id  customer_id     staff_id
count  1000.00000   1000.000000  1000.000000  1000.000000
mean    501.18000   2258.179000   296.408000     1.521000
std     289.19665   1314.667454   172.509319     0.499809
min       1.00000      2.000000     1.000000     1.000000
25%     250.75000   1157.500000   150.000000     1.000000
50%     501.50000   2244.500000   296.000000     2.000000
75%     751.25000   3371.750000   445.250000     2.000000
max    1001.00000   4581.000000   597.000000     2.000000

In [64]:
df_rental.drop(columns='last_update', inplace=True)

In [65]:
df_rental.head()

rental_id          rental_date  inventory_id  customer_id  \
0          1  2005-05-24 22:53:30           367          130   
1          2  2005-05-24 22:54:33          1525          459   
2          3  2005-05-24 23:03:39          1711          408   
3          4  2005-05-24 23:04:41          2452          333   
4          5  2005-05-24 23:05:21          2079          222   

           return_date  staff_id  
0  2005-05-26 22:04:30         1  
1  2005-05-28 19:40:33         1  
2  2005-06-01 22:12:39         1  
3  2005-06-03 01:43:41         2  
4  2005-06-02 04:33:21         1

The dataframe is good, let's export it

In [66]:
df_film.to_csv('../clean_data/rental_clean.csv', index= 0)